ci/build_wheel.sh

```diff
@@ -33,9 +33,49 @@ rapids-generate-version > ./VERSION
 
 RAPIDS_PY_CUDA_SUFFIX="$(rapids-wheel-ctk-name-gen ${RAPIDS_CUDA_VERSION})"
 
+touch /tmp/constraints-build.txt
+
+if [[ "${package_name}" == "cuspatial" ]]; then
+  # Downloads libcuspatial wheel from this current build,
+  # then ensures 'cuspatial' wheel builds always use the 'libcuspatial' just built in the same CI run.
+  RAPIDS_PY_WHEEL_NAME="libcuspatial_${RAPIDS_PY_CUDA_SUFFIX}" rapids-download-wheels-from-s3 cpp /tmp/libcuspatial_dist
+  echo "libcuspatial-${RAPIDS_PY_CUDA_SUFFIX} @ file://$(echo /tmp/libcuspatial_dist/libcuspatial_*.whl)" > /tmp/constraints-build.txt
+fi
+
+rapids-logger "Generating build requirements"
+declare -r matrix_selectors="cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch);py=${RAPIDS_PY_VERSION};cuda_suffixed=true"
+
+rapids-dependency-file-generator \
+  --output requirements \
```

Comment:
This approach means we're committing to always specifying `requirements` in `dependencies.yaml` alongside `pyproject`. I've been guilty of neglecting this in the past, and I suspect others are too. (This approach also means people may start neglecting `pyproject` and we wouldn't notice.) Plus, it means we're duplicating all of `pip`'s build dependency installation logic. Is there any way we can keep the duplication to a minimum?

Reply:
Ultimately, I think the correct approach is adding a `--build-isolation-dir` argument to `pip wheel` so that we can control the prefix for `sccache` purposes, but that obviously won't work in the short term.

Reply:
At a minimum, I think we'll have to install the build backend (`rapids-build-backend` + `setuptools` / `scikit-build-core`) separately like this for this to work, because the build backend is needed to determine all the other build-time dependencies. That should simplify the scripts and alleviate most of the "people may start neglecting `pyproject`" worry, I hope.

It should be possible to then have the rest of the build dependencies (`librmm`, `libcudf`, etc.) still be determined by `rapids-build-backend` and no require pre-installation. Like this:

```shell
# before
pip wheel --no-build-isolation --no-deps .

# after
pip wheel --no-build-isolation .
```

Let me try that.

Reply:
Yeah, all the concerns are valid but also not new. These are essentially all of the reasons why I preferred to use build isolation when setting everything up, but I don't see a way around this right now. 

> Ultimately, I think the correct approach is adding a --build-isolation-dir argument to pip wheel so that we can control the prefix for sccache purposes, but that obviously won't work in the short term.

I would like this to happen, but I think it will be an uphill battle. Repacking dynamic libraries in wheels is still something that is pretty uncommon outside of auditwheel-like bundling use cases, and repackaging headers is even rarer. I'm not saying that we shouldn't fight this battle, but I expect that it will take a long time before this changes.